src/CommandRunner.cpp

```diff
@@ -0,0 +1,204 @@
+#include "CommandRunner.hpp"
+#include <memory>
+#include <stdio.h>
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <signal.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <time.h>
+
+#include <glog/logging.h>
+#include <stout/try.hpp>
+
+#define TEMP_FILE_TEMPLATE "/tmp/criteo-mesos-XXXXXX"
+
+#define READ 0
+#define WRITE 1
+
+namespace criteo {
+namespace mesos {
+namespace CommandRunner {
+
+/*
+ * Represent a temporary file that can be either written or read from.
+ */
+class TemporaryFile
+{
+public:
+  TemporaryFile() {
+    char filepath[] = TEMP_FILE_TEMPLATE;
+    if(mkstemp(filepath) == -1)
+      throw std::runtime_error(
+        "Unable to create temporary file to run commands");
+    m_filepath = std::string(filepath);
+  }
+
+  /*
+   * Read whole content of the temporary file.
+   * @return The content of the file.
+   */
+  std::string readAll() const {
+    std::ifstream ifs(m_filepath);
```

Comment:
Even if the file was accessible during the instantiation of TemporaryFile, the file could have been deleted or even chmod... May need to check for errors here ?

Reply:
Yes, I had this in mind as well. I will implement it in another review.